In [1]:
# steps involved in univariate analysis, and understandings

In [2]:
from utils.dataparser import DataOps
data = DataOps(path="F:\dataset\marketing_data.csv")
data_transformed = data.clean_data()

data type of the given object
Income field type casted.


In [3]:
# univariate
# discrete data
d_df = data_transformed.select_dtypes(include=['object'])
con_df = data_transformed.select_dtypes(include=['int64', 'float64'])

In [4]:
d_df

Education Marital_Status Dt_Customer Country
0     Graduation       Divorced     6/16/14      SP
1     Graduation         Single     6/15/14      CA
2     Graduation        Married     5/13/14      US
3     Graduation       Together     5/11/14     AUS
4     Graduation         Single      4/8/14      SP
...          ...            ...         ...     ...
2235         PhD       Divorced      3/7/13      US
2236    2n Cycle        Married     1/22/13      SP
2237  Graduation       Divorced     12/3/12      SP
2238  Graduation        Married    11/29/12     IND
2239         PhD        Married      9/1/12      CA

[2240 rows x 4 columns]

In [5]:
con_df

ID  Year_Birth   Income  Kidhome  Teenhome  Recency  MntWines   
0      1826        1970  84835.0        0         0        0       189  \
1         1        1961  57091.0        0         0        0       464   
2     10476        1958  67267.0        0         1        0       134   
3      1386        1967  32474.0        1         1        0        10   
4      5371        1989  21474.0        1         0        0         6   
...     ...         ...      ...      ...       ...      ...       ...   
2235  10142        1976  66476.0        0         1       99       372   
2236   5263        1977  31056.0        1         0       99         5   
2237     22        1976  46310.0        1         0       99       185   
2238    528        1978  65819.0        0         0       99       267   
2239   4070        1969  94871.0        0         2       99       169   

      MntFruits  MntMeatProducts  MntFishProducts  ...  NumCatalogPurchases   
0           104              379              111  ...                    4  \
1             5               64                7  ...                    3   
2            11               59               15  ...                    2   
3             0                1                0  ...                    0   
4            16               24               11  ...                    1   
...         ...              ...              ...  ...                  ...   
2235         18              126               47  ...                    2   
2236         10               13                3  ...                    0   
2237          2               88               15  ...                    1   
2238         38              701              149  ...                    4   
2239         24              553              188  ...                    5   

      NumStorePurchases  NumWebVisitsMonth  AcceptedCmp3  AcceptedCmp4   
0                     6                  1             0             0  \
1                     7                  5             0             0   
2                     5                  2             0             0   
3                     2                  7             0             0   
4                     2                  7             1             0   
...                 ...                ...           ...           ...   
2235                 11                  4             0             0   
2236                  3                  8             0             0   
2237                  5                  8             0             0   
2238                 10                  3             0             0   
2239                  4                  7             0             1   

      AcceptedCmp5  AcceptedCmp1  AcceptedCmp2  Response  Complain  
0                0             0             0         1         0  
1                0             0             1         1         0  
2                0             0             0         0         0  
3                0             0             0         0         0  
4                0             0             0         1         0  
...            ...           ...           ...       ...       ...  
2235             0             0             0         0         0  
2236             0             0             0         0         0  
2237             0             0             0         0         0  
2238             0             0             0         0         0  
2239             1             0             0         1         0  

[2240 rows x 24 columns]

In [6]:
# use both visual and non visual together. 
# for discrete analysis we focus on ['count', 'nunique', 'unique'] and value counts.

In [9]:
data_transformed.describe().T

count          mean           std     min       25%   
ID                   2240.0   5592.159821   3246.662198     0.0   2828.25  \
Year_Birth           2240.0   1968.805804     11.984069  1893.0   1959.00   
Income               2216.0  52247.251354  25173.076661  1730.0  35303.00   
Kidhome              2240.0      0.444196      0.538398     0.0      0.00   
Teenhome             2240.0      0.506250      0.544538     0.0      0.00   
Recency              2240.0     49.109375     28.962453     0.0     24.00   
MntWines             2240.0    303.935714    336.597393     0.0     23.75   
MntFruits            2240.0     26.302232     39.773434     0.0      1.00   
MntMeatProducts      2240.0    166.950000    225.715373     0.0     16.00   
MntFishProducts      2240.0     37.525446     54.628979     0.0      3.00   
MntSweetProducts     2240.0     27.062946     41.280498     0.0      1.00   
MntGoldProds         2240.0     44.021875     52.167439     0.0      9.00   
NumDealsPurchases    2240.0      2.325000      1.932238     0.0      1.00   
NumWebPurchases      2240.0      4.084821      2.778714     0.0      2.00   
NumCatalogPurchases  2240.0      2.662054      2.923101     0.0      0.00   
NumStorePurchases    2240.0      5.790179      3.250958     0.0      3.00   
NumWebVisitsMonth    2240.0      5.316518      2.426645     0.0      3.00   
AcceptedCmp3         2240.0      0.072768      0.259813     0.0      0.00   
AcceptedCmp4         2240.0      0.074554      0.262728     0.0      0.00   
AcceptedCmp5         2240.0      0.072768      0.259813     0.0      0.00   
AcceptedCmp1         2240.0      0.064286      0.245316     0.0      0.00   
AcceptedCmp2         2240.0      0.013393      0.114976     0.0      0.00   
Response             2240.0      0.149107      0.356274     0.0      0.00   
Complain             2240.0      0.009375      0.096391     0.0      0.00   

                         50%       75%       max  
ID                    5458.5   8427.75   11191.0  
Year_Birth            1970.0   1977.00    1996.0  
Income               51381.5  68522.00  666666.0  
Kidhome                  0.0      1.00       2.0  
Teenhome                 0.0      1.00       2.0  
Recency                 49.0     74.00      99.0  
MntWines               173.5    504.25    1493.0  
MntFruits                8.0     33.00     199.0  
MntMeatProducts         67.0    232.00    1725.0  
MntFishProducts         12.0     50.00     259.0  
MntSweetProducts         8.0     33.00     263.0  
MntGoldProds            24.0     56.00     362.0  
NumDealsPurchases        2.0      3.00      15.0  
NumWebPurchases          4.0      6.00      27.0  
NumCatalogPurchases      2.0      4.00      28.0  
NumStorePurchases        5.0      8.00      13.0  
NumWebVisitsMonth        6.0      7.00      20.0  
AcceptedCmp3             0.0      0.00       1.0  
AcceptedCmp4             0.0      0.00       1.0  
AcceptedCmp5             0.0      0.00       1.0  
AcceptedCmp1             0.0      0.00       1.0  
AcceptedCmp2             0.0      0.00       1.0  
Response                 0.0      0.00       1.0  
Complain                 0.0      0.00       1.0

What does it tells.
- ID we can ignore 
- Year-Birth
    The data focus on the 1893 to 1996. only answers the specific time range.
    Need to get understanding whether we have enough data from each year. missing or not
- Income:
    we have missing values. ? handling it.
    high std, and max is way over than 75% of the data . outlier ? how to handle.
- Kidhome/Teenhome.
    Most households have no or one child or teenager at home, with very few having two.


- MntWines/ MntFruits
- MntFishProducts/ MntMeatProducts
- MntGoldProds/ MntSweetProducts
    - +ve skewness of data, very few with high amount 

- NumDealsPurchases/ NumWebPurchases
    - we have outliers.
    

In [10]:
data_transformed.describe(include=['object']).T

count unique         top  freq
Education       2240      5  Graduation  1127
Marital_Status  2240      8     Married   864
Dt_Customer     2240    663     8/31/12    12
Country         2240      8          SP  1095

In [11]:
# Visual analysis.